### DISCUSSION TITLE
How to read data from a solidity event in python ?

### DISCUSSION THREAD

**Discussion by 0xusmanf** [smartcontractkit/full-blockchain-solidity-course-py#1871] [Q&A] [answered]
Hello everyone,

First let me explain code.

**Below is the event**

`event SubscriptionCreated(uint64 indexed subId, address owner);`

**And here is the function where its been emitted**

```
function createSubscription() external override returns (uint64 _subId) {
        s_currentSubId++;
        s_subscriptions[s_currentSubId] = Subscription({
            owner: msg.sender,
            balance: 0
        });
        emit SubscriptionCreated(s_currentSubId, msg.sender);
        return s_currentSubId;
    }
```
What I am trying to do is to read `subId` from this event, below is the python code.

```
vrf_coordinator = VRFCoordinatorV2Mock.deploy(
        config["networks"][network.show_active()]["fee"],
        GAS_PRICE_LINK,
        {"from": account},
    )
    tx = vrf_coordinator.createSubscription()
    tx.wait(1)
    SUB_ID = tx.events[0]
    print(SUB_ID)
```

first I am deploying the `VRFCoordinatorV2Mock` contract then I am calling the function `createSubscription()` where this event is being emitted, I only know how to access the entire event but dont know how to fetch subId from this event.

**Last print statement prints the following result**

`OrderedDict([('subId', 2), ('owner', '0x66aB6D9362d4F35596279692F0251Db635165871')])`

**Comment by 0xusmanf** [comment, marked as answer]
just found out the solution, you can use 1 of the 2 following ways.

`SUB_ID = tx.events["SubscriptionCreated"]["subId"]`
**or**
`SUB_ID = tx.return_value`

Thanks to @neftyr for implementing VRF V2, I found out the solution from his code [here](https://github.com/Neftyr/Lottery-On-VRFConsumerBaseV2-Python/blob/main/scripts/run_lottery.py).

**Reply by Niferu** [reply]
@usmanfarooq91 

I'm very glad it helped you! Cheers mate! :)
